app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorEffectHandler.kt

```diff
@@ -35,9 +38,22 @@ class SyncIndicatorEffectHandler @AssistedInject constructor(
           .addTransformer(FetchDataForSyncIndicatorState::class.java, fetchDataForSyncIndicatorState())
           .addTransformer(StartSyncedStateTimer::class.java, startTimer())
           .addTransformer(InitiateDataSync::class.java, startDataSync())
+          .addTransformer(FetchPendingSyncRecordCount::class.java, fetchPendingSyncCount())
           .addConsumer(ShowDataSyncErrorDialog::class.java, { uiActions.showErrorDialog(it.errorType) }, schedulersProvider.ui())
           .build()
 
+  private fun fetchPendingSyncCount(): ObservableTransformer<FetchPendingSyncRecordCount, SyncIndicatorEvent> {
+    return ObservableTransformer { effect ->
+      effect
+          .map {
+            frequentlySyncingRepositories
+                .map { it.pendingSyncRecordCountImmediate() }
```

Comment:
I think we had left it as an `Observable` here because we wanted the indicator to turn into the `Pending` state if any record on the database gets marked as `PENDING_SYNC`. In most cases, since we have to navigate away from this screen in order to modify a record and navigate back (in which case, the pending state will be queried again), it should be fine.

One case that I am concerned about though, is that this screen runs alongside the `OverdueScreen` in tabs, and that screen modifies `Appointment` records. Not implementing this in a reactive fashion might mean that this never changes to the `Pending` state even though there are pending records.

Please verify?

Reply:
Good point. I will verify this.

Reply:
This does not seem to be working before as well. I will fix this in the next PR.
Is that ok?

Reply:
Yes, let's fix this later then. If it's not an easy fix, can you raise a ticket for this instead? Someone else can pick this up since you'll be rolling off soon.